lib/apiservers/engine/backends/container.go

```diff
@@ -592,8 +592,18 @@ func (c *Container) ContainerCreate(config types.ContainerCreateConfig) (contain
 	}
 
 	// Reserve the container name to prevent duplicates during a parallel operation.
-	if err := cache.ContainerCache().ReserveName(container, config.Name); err != nil {
-		return containertypes.ContainerCreateCreatedBody{}, derr.NewRequestConflictError(err)
+	if config.Name != "" {
+		err := cache.ContainerCache().ReserveName(container, config.Name)
+		if err != nil {
+			return containertypes.ContainerCreateCreatedBody{}, derr.NewRequestConflictError(err)
+		}
+	} else {
+		for {
+			config.Name = namesgenerator.GetRandomName(0)
+			if cache.ContainerCache().ReserveName(container, config.Name) == nil {
```

Comment:
We should add a test case in the `1-04-Docker-Create` suite for this. There's already a test at https://github.com/vmware/vic/blob/master/tests/test-cases/Group1-Docker-Commands/1-04-Docker-Create.robot#L194 - we can add this case there.

Reply:
I am unaware of a means to manipulate urandom to force a collision - I will look at adding a unit test and mocking `GetRandomName`.

Reply:
Oh right, my bad!